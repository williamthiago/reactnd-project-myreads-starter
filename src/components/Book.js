import React from 'react'
import PropTypes from 'prop-types'

import bookPropType from './bookPropType'
import shelfType from '../enums/shelfType'
import BookCover from './BookCover'
import noCover from '../icons/no-cover.png'

const Book = ({ book, onChangeShelf }) => {
  let { authors, title, shelf, imageLinks } = book
  authors = authors ? authors.join('; ') : 'Author Unknow'
  
  let backgroundImage = imageLinks ? imageLinks.thumbnail : noCover

  const onChange = (event) => {
    let shelf = event.target.value
    onChangeShelf && onChangeShelf(book, shelf)
  } 
  
  return (
    <div className="book">
      <div className="book-top">
        <BookCover image={backgroundImage} />
        <div className="book-shelf-changer">
          <select defaultValue={shelf} onChange={onChange} >
            <option value="none" disabled>Move to...</option>
            {Object.keys(shelfType).map(type => (
              <option key={type} value={type}>{shelfType[type]}</option>
            ))}
            <option value="none">None</option>
          </select>
        </div>
      </div>
      <div className="book-title">{title}</div>
      <div className="book-authors">{authors}</div>
    </div>
  )
}

Book.propTypes = {
  book: bookPropType.isRequired,
  onChangeShelf: PropTypes.func.isRequired
}

export default Book