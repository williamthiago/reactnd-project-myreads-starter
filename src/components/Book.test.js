import React from 'react'
import { shallow } from 'enzyme'

import Book from './Book'

it('Renders Book without crashing', () => {
  const onChangeShelf = () => {}
  const book = sampleBooks[0]
  expect(shallow(<Book onChangeShelf={onChangeShelf} book={book} />)).toMatchSnapshot()
})