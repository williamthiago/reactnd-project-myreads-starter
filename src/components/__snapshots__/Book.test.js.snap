// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Book without crashing 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <BookCover
      image="http://books.google.com/books/content?id=evuwdDLfAyYC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
    />
    <div
      className="book-shelf-changer"
    >
      <select
        defaultValue="wantToRead"
        onChange={[Function]}
      >
        <option
          disabled={true}
          value="none"
        >
          Move to...
        </option>
        <option
          value="currentlyReading"
        >
          Currently Reading
        </option>
        <option
          value="wantToRead"
        >
          Want To Read
        </option>
        <option
          value="read"
        >
          Read
        </option>
        <option
          value="none"
        >
          None
        </option>
      </select>
    </div>
  </div>
  <div
    className="book-title"
  >
    The Cuckoo's Calling
  </div>
  <div
    className="book-authors"
  >
    Robert Galbraith
  </div>
</div>
`;
